### ISSUE TITLE
crash android application from attach->enumerate->detach

### ISSUE THREAD

**Issue by juliangruber** [frida/frida-core#49]
This script using frida-node will reliably crash any app on my 5.0 device, presumably when detaching from it:

``` js
var frida = require('frida');

console.log('get device');
frida.getUsbDevice()
.then(function(device){

  console.log('get frontmost application');
  return device.getFrontmostApplication()
  .then(function(app){
    console.log('attach');
    return device.attach(app.pid)
  })
  .then(function(session){
    console.log('enumerate');
    return session.enumerateRanges('rw-')
    .then(function(){
      console.log('detach');
      return session.detach();
    });
  });
})
.catch(function(err){
  console.error(err);
});
```

When `session.enumerateRanges('rw-')` is left out, it doesn't crash.

**Comment by juliangruber** [comment]
replacing `session.enumerateRanges` with a 1 second timeout doesn't trigger the error

**Comment by oleavr** [comment]
Nice catch! That's indicative of it being a problem during deinitialization after a script has been created (which `session.enumerateRanges()` creates behind the scenes). I'll have a look.

**Comment by oleavr** [comment]
Fixed in latest Frida.
